zerver/views/auth.py

```diff
@@ -210,6 +216,10 @@ def google_oauth2_csrf(request, value):
 def start_google_oauth2(request):
     # type: (HttpRequest) -> HttpResponse
     url = reverse('zerver.views.auth.send_oauth_request_to_google')
+
+    if not (settings.GOOGLE_OAUTH2_CLIENT_ID and settings.GOOGLE_OAUTH2_CLIENT_SECRET):
```

Comment:
I think if we add this check in `zerver.views.registration.accounts_home` and show the error in `templates.zerver.accounts_home`, we can warn the user and administrator before the user initiates the process of login/signup.